### ISSUE TITLE
Issue with Chat Vision

### ISSUE THREAD

**Issue by GitHubUsername1** [RageAgainstThePixel/com.openai.unity#158]
# Bug Report
Might not be a bug, could just be me (probably is, I've tried to solve it myself, no luck).

## Overview
I'm having an issue with Chat Vision, the response message is cut off. I went over the readme and seen this is due to the response generating unending whitespace because of not specifying in the system message the response should be Json, and also the enum formatting type should be Json, so I changed my code. However now I'm getting the error -  "error": {
    "message": "Invalid content type. image_url is only supported by certain models.",
    "type": "invalid_request_error",
    "param": "messages.[1].content.[1].type",
    "code": null
In the console.

## To Reproduce
The code to create a response with unending whitespace in response -

`var messages = new List<OpenAI.Chat.Message>
    {
        new OpenAI.Chat.Message(Role.System, "You are a helpful assistant."),
        new OpenAI.Chat.Message(Role.User, new List<OpenAI.Chat.Content>
        {
             photoPrompt,
             texture
        })
    };
    var chatRequest = new ChatRequest(messages, model: "gpt-4-vision-preview");
    var response = await apiVision.ChatEndpoint.GetCompletionAsync(chatRequest);`

This generated no error, the image was processed, however the response was cut off.
So I changed my code to ask for response in JSON format and to specify the enum formatting (ChatResponseFormat) to be JSON -

`var apiVision = new OpenAIClient(openaikey);
var messages = new List<OpenAI.Chat.Message>
{
    new OpenAI.Chat.Message(Role.System, "You are a helpful assistant designed to output JSON."),
    new OpenAI.Chat.Message(Role.User, new List<OpenAI.Chat.Content>
    {
        photoPrompt,
        texture 
    })
 
};

var chatRequest = new ChatRequest(messages, "gpt-4-1106-preview", responseFormat: ChatResponseFormat.Json);
//var chatRequest = new ChatRequest(messages, model: "gpt-4-vision-preview");
var response = await apiVision.ChatEndpoint.GetCompletionAsync(chatRequest);`

## Expected behavior
I don't understand the error as I'm not passing in an image url (although I have tried it).

## Screenshots
![Capture](https://github.com/RageAgainstThePixel/com.openai.unity/assets/6526207/b046aec5-301d-400b-9fc0-060a77a75b75)

## Additional context
In summary - The image was accepted and a response received when not specifying Json, but had unending whitespace in response and response was truncated. If I specify Json the api call fails with 'Invalid content type. image_url is only supported by certain models'.

**Comment by StephenHodgson** [comment]
Yeah this is pretty strange. I would try to do this in the OpenAI Playground as a start to see if it is even possible. 

It could be that the vision-preview model just doesn't support BOTH vision and JSON output modes.

**Comment by StephenHodgson** [comment]
> I don't understand the error as I'm not passing in an image url (although I have tried it).

You are passing an image url, since that `texture` input is being encoded to base64, and ImageUrl base type.

**Comment by StephenHodgson** [comment]
Have you tried the same code without the JSON formatting?
